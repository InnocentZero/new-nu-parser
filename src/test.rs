use crate::resolver::Resolver;
use crate::{compiler::Compiler, parser::Parser};
use std::path::Path;
use std::process::exit;

fn evaluate_example(fname: &Path) -> String {
    let mut compiler = Compiler::new();
    let contents = std::fs::read(fname).expect("We only run tests found by glob");

    let span_offset = compiler.span_offset();
    compiler.add_file(&fname.to_string_lossy(), &contents);

    let parser = Parser::new(compiler, span_offset);
    compiler = parser.parse();

    let mut result = compiler.display_state();

    if !compiler.errors.is_empty() {
        return result;
    }

    let mut resolver = Resolver::new(&compiler);
    resolver.resolve();
    result.push_str(&resolver.display_state());

    result
}

#[test]
fn test_node_output() {
    insta::glob!("../tests", "*.nu", |path| {
        insta::assert_snapshot!(evaluate_example(path));
    });
}
