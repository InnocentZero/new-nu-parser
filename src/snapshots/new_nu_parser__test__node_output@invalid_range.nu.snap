---
source: src/test.rs
expression: evaluate_example(path)
input_file: tests/invalid_range.nu
---
0: Int (0 to 1)
1: Garbage (2 to 4)
2: Int (5 to 6)
3: Block(BlockId(0)) (0 to 7)
==== ERRORS ====
Error (NodeId 1): incomplete expression
==== SCOPE ====
0: Frame Scope, variables: [  ], node_id: NodeId(3)

